File: backend/app/schemas/signal.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class AnalysisSignalBase(BaseModel):
    instrument_id: int
    timeframe: str
    signal_type: str
    direction: str
    strength: str | None = None
    generated_at: datetime
    expires_at: datetime | None = None
    payload: dict[str, Any] | None = None


class AnalysisSignalCreate(AnalysisSignalBase):
    pass


class AnalysisSignalRead(AnalysisSignalBase):
    model_config = ConfigDict(from_attributes=True)

    id: int

File: frontend/src/types/signal.ts
export type AnalysisSignal = {
  id: number;
  instrument_id: number;
  timeframe: string;
  signal_type: string;
  direction: string;
  strength: string | null;
  generated_at: string;
  expires_at: string | null;
  payload: Record<string, unknown> | null;
};
